util/argo/argo.go

```diff
@@ -125,6 +159,22 @@ func FilterByName(apps []argoappv1.Application, name string) ([]argoappv1.Applic
 	return items, status.Errorf(codes.NotFound, "application '%s' not found", name)
 }
 
+// FilterByNameP returns pointer applications
+// This function is for the changes in #12985.
+func FilterByNameP(apps []*argoappv1.Application, name string) ([]*argoappv1.Application, error) {
+	if name == "" {
+		return apps, nil
+	}
+	items := make([]*argoappv1.Application, 0)
+	for i := 0; i < len(apps); i++ {
+		if apps[i].Name == name {
+			items = append(items, apps[i])
+			return items, nil
+		}
+	}
+	return items, status.Errorf(codes.NotFound, "application '%s' not found", name)
```

Comment:
One thing we have to be careful of is revealing the existence or non-existence of an Application before establishing whether the user is allowed to know that info.

I think in the case of this error, we should probably just return an empty list rather than an error.